.github/workflows/ci_cd.yml

```diff
@@ -822,12 +822,24 @@ jobs:
       - name: Run Playwright tests
         run: just frontend/run ${{ matrix.script }} --workers=2
 
+      - name: Check for changes after running tests with -u
+        id: check_snapshots_changed
+        run: |
+          # shellcheck disable=SC2086
+          if [ -n "$(git diff --binary --name-only)" ]; then
+            git add . && git diff --staged --binary | gzip -f > /tmp/snapshot_diff.patch.gz
+            echo "snapshots_changed=true" >> $GITHUB_OUTPUT
+            git diff --binary --name-only
+          else
+            echo "snapshots_changed=false" >> $GITHUB_OUTPUT
+          fi
+
       - uses: actions/upload-artifact@v4
-        if: failure()
+        if: failure() || steps.check_snapshots_changed.outputs.snapshots_changed == 'true'
         id: test-results
         with:
-          name: ${{ matrix.name }}_test_results
-          path: frontend/test-results
```

Comment:
We should still upload the test-results file. The trace can still be useful for non-snapshot tests that fail, especially to debug if they fail in CI but pass locally. Would it still get uploaded by some other means I'm not understanding?

Reply:
I agree. So what I'm picking from this is that we should upload both the trace like we already do, alongside the patches(if they exist)?

Reply:
Yep :sligh